### ISSUE TITLE
memcacheExtensionConfigured.policy.yml should check PHP version before Failing

### ISSUE THREAD

**Issue by markf3lton** [fiasco/drutiny-plugin-drupal-8#10]
There are currently 3 memcache checks for drutiny - 
memcacheDefaultCacheBackend
memcacheEnabled
memcacheExtensionConfigured

The memcacheExtensionConfigured policy checks that memcached is configured as the default backend.

```
# Drupal 8 $settings['memcache']['extension'] = 'Memcached';
```

See Step 7 of https://docs.acquia.com/acquia-cloud/performance/memcached/

### The problem

I believe PECL extension "memcached" only needs to be specified in settings.php if the site is using PHP 5.6. 

I have D8 sites failing this Audit yet Memcached is working fine (because memcacheEnabled = Pass and memcacheDefaultCacheBackend = Pass).

When PHP 7.1 becomes the default later this year, this Policy may be throwing false positives? 

### Possible fix

Maybe we can build a check to determine if the site is running PHP 5.6 or PHP 7.1. If so, the memcacheExtensionConfigured policy could depend on it.

**Comment by markf3lton** [comment]
Related:  https://github.com/drutiny/plugin-drupal-8/pull/11

**Comment by markf3lton** [comment]
@shanefjordan  for viz

**Comment by fiasco** [comment]
Audit classes have the ability to require certain conditions before evaluating a policy (https://drutiny.github.io/2.1.x/audits/#audit-prerequisites) which is where it would make sense whether this particular check was relevant but there are two issues with the architecture currently that block this:
1. The check for the memcache extension uses a generic Drupal 8 [SettingsCompare](https://drutiny.github.io/2.1.x/audits/DrutinyPluginDrupal8Audit/#settingcompare) Audit by which it wouldn't make sense to add a specific prerequisite for < PHP 7.
2. The policies that a policy depends on, are added to the audit before the prerequisites are checked which doesn't make sense. Not a big issue in this case, but we'd want to fix that up while we're at it.

I do have a solution in mind:

- Build a `requireDrupal8ConditionalEnvironment` prerequisite method in a PHP trait
- Utilise the Symfony expression language to evaluate if `requireDrupal8ConditionalEnvironment` returns `TRUE` or `FALSE`. 
- Allow the expression to be defined by the Policy metadata so it can be customised to each policy.

**Comment by fiasco** [comment]
Have fixed this in 2.3.x: https://github.com/drutiny/content/blob/master/Policy/drupal-8/memcacheExtensionConfigured.policy.yml

Read more on dependencies here: https://drutiny.github.io/2.3.x/policy/#depends
